File: back/src/main/java/com/pingo/entity/keywords/KeywordGroup.java
package com.pingo.entity.keywords;

import lombok.Getter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;
@Getter
@ToString
public class KeywordGroup {
    private String kwId;
    private String kwName;
    private String kwMessage;
    private List<Keyword> childKeyword;

    public KeywordGroup(String kwId, String kwName, String kwMessage) {
        this.kwId = kwId;
        this.kwName = kwName;
        this.kwMessage = kwMessage;
        this.childKeyword = new ArrayList<>();
    }

    public void addChildKeyword(Keyword keyword) {
        if (this.kwId.equals(keyword.getKwParentId())) {
            this.childKeyword.add(keyword);
        }
    }
}

File: front/lib/data/models/keyword_model/keyword_group.dart
import 'keyword.dart';

class KeywordGroup {
  String? kwId;
  String? kwName;
  String? kwMessage;
  List<Keyword>? childKeyword;

  KeywordGroup(
      {required this.kwId,
      required this.kwName,
      required this.kwMessage,
      required this.childKeyword});

  KeywordGroup.fromJson(Map<String, dynamic> json)
      : kwId = json['kwId'],
        kwName = json['kwName'],
        kwMessage = json['kwMessage'],
        childKeyword = (json['childKeyword'] as List<dynamic>?)
            ?.map((child) => Keyword.fromJson(child as Map<String, dynamic>))
            .toList();
}
